pkg/catalogsource/deletion.go

```diff
@@ -0,0 +1,136 @@
+package catalogsource
+
+import (
+	"context"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/builders"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// NewDeletionReconciler returns a new GrpcDestroyer built from the provided parameters.
+func NewDeletionReconciler(log *logrus.Entry, client client.Client) DeletionReconciler {
+	return DeletionReconciler{
+		log:    log,
+		client: client,
+	}
+}
+
+// DeletionReconciler encapsulates the logic used to remove all resources related to a registry.
+type DeletionReconciler struct {
+	log    *logrus.Entry
+	client client.Client
+}
+
+// DeleteRegistryResources deletes all resources that make up a registry.
+func (r *DeletionReconciler) DeleteRegistryResources(ctx context.Context, name, namespace, targetNamespace string) (err error) {
```

Comment:
Same comment as before. You are deleting a `CatalogSource` and all its resources.